dashboard-frontend/src/hooks/useAuth0.tsx

```diff
@@ -0,0 +1,33 @@
+import { useAuth0 as useRealAuth0 } from '@auth0/auth0-react';
+
+const useAuth0 = (): Pick<
+  ReturnType<typeof useRealAuth0>,
+  | 'isAuthenticated'
+  | 'isLoading'
+  | 'user'
+  | 'logout'
+  | 'getAccessTokenSilently'
+  | 'loginWithRedirect'
+> => {
+  const auth0Context = useRealAuth0();
+
+  if (process.env.REACT_APP_RUNNING_IN_E2E_MODE) {
```

Comment:
Really don't love this. It's a mock. But instead of the testing program being in charge of creating the mock or the app having some sort of inversion of control that makes it easy to mock, the app code itself checks for an env variable and then returns a mock! 

:/

Reply:
Yes I agree with you that this is not ideal.

The thing is that the app runs on a separate process than Cypress, so out of the box is not possible to mock a module.

This mean that in case you want to change the default implementation of something during Cypress tests, you'll need to instrument the code always.

Right now the fake implementation that is used when Cypress tests are running is hardcoded, but this control can be given to Cypress if needed. Instead of returning this hardcoded object, the app code during Cypress tests could read a property from the Windows object which Cypress is able to modify to provide the right mock.

At the moment I don't mind having it like this, since I want to have this auth state for all the e2e tests, in the future when we update some of the e2e tests to talk to the real backend, we can think about improving this.

Before going with this approach I did some research and the Cypress team is already working on an interesting solution for this. It's still in alpha but I'm watching they git repo for updates. Take a look at this when you can:
- https://www.youtube.com/watch?v=RAFdYqRO2vI
	- https://github.com/bahmutov/mock-in-bundle